psignifit/_configuration.py

```diff
@@ -131,16 +131,41 @@ def check_fixed_parameters(self, value):
             vkeys = _PARAMETERS
             if vkeys < set(value.keys()):
                 raise PsignifitException(
-                    f'Option fixed_paramters keys must be in {vkeys}. Given {list(value.keys())}!'
+                    f'Option fixed_parameters keys must be in {vkeys}. Given {list(value.keys())}!'
                 )
-
+            # Check that the fixed parameters have been set to meaningful values
+            if (gamma := value.get('gamma')) and (lambda_ := value.get('lambda')):
```

Comment:
in these `if`s, I think you want to check if the key has been defined. You need to use `if 'gamma' in value` instead.

For example, if `value = {'gamma': 0.0}`, `value.get('gamma')` is 0.0 and so evaluated as False.

In this snippet, the print statement will not be executed:
```
value = {'gamma': 0.0}
if value.get('gamma'): print('alright')
```

Reply:
Case where it would fail: gamma=0, lambda=1 would not raise an exception

Reply:
I added two tests in 5fc7e8410c6d53945795fdca0f3ed07786189c63 which show that the correct `Exception` is thrown.

It is true that the code in the if-block is not executed if one of the two parameters is exactly zero, but in that case the code below will make sure that the other, non-zero, parameters is strictly smaller than 1, so at the end everything works as expected. I could change the comment in the if to make it more clear?

Or make the code more verbose as you propose... I don't know. Somehow I think it's nicer that if the input is `gamma=0, lambda=2` the error complains about `lambda` alone and not about the sum of `gamma + lambda`, but it's a matter of taste...

What do you think?

Reply:
Yes, the code still works ... as a side effect

My issue is in the meaning of the if test. `if value.get('gamma')` means "if the value of value['gamma'] evaluates to False (and if the key 'gamma' is missing, that's the case)". `if 'gamma' in value` means "if the key 'gamma' is defined).

Then if you want to have an error about lambda alone in case `gamma=0, lambda=2`, that can also be arranged by checking the individual keys first. It is a separate concern

It's a matter of having the code more transparent, please leave it like it if you think I'm nitpicking. The code works.

Reply:
fixed in 7280b470a87052d2562565aa4589bddd824dbb9c